src/transformers/trainer.py

```diff
@@ -1363,7 +1363,18 @@ def train(
             # We load the model state dict on the CPU to avoid an OOM error.
             state_dict = torch.load(os.path.join(self.state.best_model_checkpoint, WEIGHTS_NAME), map_location="cpu")
             # If the model is on the GPU, it still works!
-            self.model.load_state_dict(state_dict)
+            load_result = self.model.load_state_dict(state_dict, strict=False)
+            if len(load_result.missing_keys) != 0:
+                if load_result.missing_keys == self.model._keys_to_ignore_on_save:
```

Comment:
btw, does it make sense to compare sets instead of arrays? (e.g. `set(load_result.missing_keys).issubset(set(self.model._keys_to_ignore_on_save))`). Or we are sure the order is always the same?

Reply:
There is the same tests for all models in the CI, so this works for all of the Transformers models.

Nevertheless, your tests is nicer so feel free to update with it. Also could you put this code in a private method (so we don't have it twice)? Or I can refactor after this PR, as you prefer.

Reply:
@sgugger , I've extracted code duplication in the private method and updated the check to comparing sets. Unfortunately, I did not find any unit tests related to loading the best model, so I did not add a check. You are welcomed to update this PR if it needs improvements